### PR TITLE
Fix windows path translation problem

### PR DESCRIPTION
**Pull request by richtera**
When using path.join under windows the system will use backslashes which won't work with zookeeper paths.

### CODE REVIEW

File: session.js [hunk 1]
```diff
@@ -356,7 +356,7 @@ module.exports = function (
 	// internal API
 
 	Session.prototype._chroot = function (path) {
-		path = paths.join(this.root, path)
+		path = paths.join(this.root.replace('/', path.sep), path.replace('/', path.sep)).replace(path.sep, '/');
```
Comment (dannycoates) [inline session.js]: Interesting, I didn't know about the different `path.join` behavior on Windows, but I guess it makes sense. This looks good except that node's path module is imported as `paths` (with an 's'). If you could change that I'll get this merged in.

Comment (richtera) [inline session.js]: There is a better fix actually in my branch.
Making a new pull.
Andy

**Comment by richtera** [comment]
This is working now. Rather than trying to fix path.join by replacing \ with / just implement what path.join does.

**Comment by dannycoates** [comment]
Are you a regular Windows user? I'm looking for someone to keep this project working on Windows. If so I offer you commit access :)

**Comment by richtera** [comment]
I using windows, linux (ubuntu) and mac. It's working so far

Sent from my iPhone

On Apr 25, 2013, at 12:23 PM, Danny Coates notifications@github.com wrote:

Are you a regular Windows user? I'm looking for someone to keep this
project working on Windows. If so I offer you commit access :)

—
Reply to this email directly or view it on
GitHubhttps://github.com/dannycoates/zkjs/pull/7#issuecomment-17017731
.

**Pull request merged**